api_v3/services/UserService.php

```diff
@@ -629,6 +629,102 @@ public function enableLoginAction($userId, $loginId, $password = null)
 		return $this->getResponseUserWithEncryptedSeed($user);
 	}
 
+	/**
+	 * Replace a user's existing login data to a new or an existing login data on a different partner while keeping the existing password.
+	 * to only be used when admin impersonates a partner
+	 *
+	 * @action replaceUserLoginData
+	 *
+	 * @param string $userId The user's unique identifier in the partner's system
+	 * @param string $newLoginId The new user's email address that identifies the user for login
+	 * @param string $existingLoginId The user's email address that identifies the user for login
+	 * @return KalturaUser The user object represented by the user and login IDs
+	 * @maskedParams userId,newLoginId,currentLoginId
+	 *
+	 * @throws KalturaErrors::USER_NOT_FOUND
+	 * @throws KalturaErrors::ADMIN_LOGIN_USERS_QUOTA_EXCEEDED
+	 * @throws KalturaErrors::LOGIN_ID_ALREADY_USED
+	 * @throws KalturaErrors::USER_LOGIN_ALREADY_ENABLED
+	 * @throws KalturaErrors::USER_DATA_ERROR
+	 * @throws KalturaErrors::LOGIN_DATA_NOT_PROVIDED
+	 * @throws KalturaErrors::LOGIN_DATA_MISMATCH
+	 * @throws KalturaErrors::CANNOT_DISABLE_LOGIN_FOR_ADMIN_USER
+	 * @throws KalturaErrors::USER_LOGIN_ALREADY_DISABLED
+	 * @throws KalturaErrors::LOGIN_ID_ALREADY_USED
+	 */
+	public function replaceUserLoginDataAction($userId, $newLoginId, $existingLoginId = null)
```

Comment:
asked Gonen the exact question. he's reply:
"EP will make the necessary API calls and would know if the user has login ID or not, but it will be using this API for both users that do not have loginId and users that do, that’s why the parameter needs to be optional."